python/vowpalwabbit/pyvw.py

```diff
@@ -709,9 +707,9 @@ def predict(self, ec, prediction_type: Optional[Union[int, PredictionType]] = No
             prediction_type = self.get_prediction_type()
 
         if isinstance(ec, Example):
-            prediction = get_prediction(ec, prediction_type)
+            prediction = ec.get_prediction(prediction_type)
         else:
-            prediction = get_prediction(ec[0], prediction_type)
+            prediction = ec[0].get_prediction(prediction_type)
```

Comment:
nit: `elif isinstance(ec, List)` to make it clear what this is looking for

Reply:
On line 692 the type has already been checked to be example or list. Adding another type check here seems redundant